packages/starksheet-webapp/src/hooks/useChainProvider.ts

```diff
@@ -0,0 +1,5 @@
+import { starknetChainProvider } from "../provider";
+
+export function useChainProvider() {
```

Comment:
declares a useChainProvider hook, that just resends the starknetChainProvider.

This first brick allowed me to check that the evmProvider implementation matches the code expectations and compiles.

The code still not runs properly with an evm provider because there are other ties to starknet, which will be addressed in subsequent PRs